test/topology_custom/test_change_replication_factor_1_to_0.py

```diff
@@ -65,3 +66,52 @@ async def do_reads() -> None:
     await asyncio.sleep(1)
     stop_event.set()
     await asyncio.gather(*tasks)
+
+# Tests #22688 - we should be able to both do further alter:s of a keyspace
+# even after removing replication factor fully from a dc and decommission of said
+# dc.
+@pytest.mark.parametrize(
+    "use_tablets",
+    [
+        pytest.param(False, id="vnodes"),
+        pytest.param(True, id="tablets"),
+    ],
+)
+@pytest.mark.asyncio
+async def test_change_replication_factor_1_to_0_and_decommission(request: pytest.FixtureRequest, manager: ManagerClient, use_tablets: bool) -> None:    
+    CONFIG = {"endpoint_snitch": "GossipingPropertyFileSnitch", "enable_tablets": str(use_tablets)}
+    logger.info("Creating a new cluster")
+    for i in range(2):
+        await manager.server_add(
+            config=CONFIG,
+            property_file={'dc': f'dc{i}', 'rack': f'myrack{i}'})
+
+    cql = manager.get_cql()
+    await cql.run_async("create keyspace ks with replication = {'class': 'NetworkTopologyStrategy', 'dc0': 1, 'dc1': 1}")
+    await cql.run_async("create table ks.t (pk int primary key)")
+
+    srvs = await manager.running_servers()
+    sorted(srvs, key=lambda si: si.datacenter)
+    assert(srvs[1].datacenter == "dc1")
+
+    await wait_for_cql_and_get_hosts(cql, srvs, time.time() + 60)
+
+    keys = range(256)
+    await asyncio.gather(*[cql.run_async(f"INSERT INTO ks.t (pk) VALUES ({k});") for k in keys])
+
+    # dc1 = 0 -> remove me from said dc
+    await cql.run_async("alter keyspace ks with replication = {'class': 'NetworkTopologyStrategy', 'dc0': 1, 'dc1': 0}")
+
+    logger.info(f"Decommissioning node {srvs[1]}")
+    
+    # decommission dc1
+    await manager.decommission_node(srvs[1].server_id)
+    await check_token_ring_and_group0_consistency(manager)
+
+    # ensure this no-op alter still works
+    async with asyncio.timeout(30):
+        await cql.run_async("alter keyspace ks with replication = {'class': 'NetworkTopologyStrategy', 'dc0': 1}")
+
+    # ensure we can stop the servers nicely
+    await manager.server_stop_gracefully(srvs[1].server_id)
```

Comment:
nit: can gather_io to stop both in parallel.
But why is that part of the test at all?

Reply:
I want to be sure shutdown is ok. Not really required, but there are occasionally issues manifesting on "full flush".

Reply:
There occasional issues manifesting in other cases as well, like bootstrapping a new node, say in a new dc, and then altering the keyspace to set the rf in that dc, which seems a more relevant case that this doesn't test.

If it's just paranoia around "full flush" and you haven't encountered a relevant issue related to this change then we should drop it.  It's just bloat (if all tests need to gracefully stop all nodes, then this should be done by the framework like we do in dtest.  I just don't want random tests do that with no explicit reason)

As for the case I described above.  I think it's worthwhile adding, isn't it?

Reply:
Fair enough. Testing adding a dc sounds like a separate test.